src/server/scripts/EasternKingdoms/Stratholme/instance_stratholme.cpp

```diff
@@ -119,6 +166,21 @@ class instance_stratholme : public InstanceMapScript
                         }
                         break;
                     }
+                    case NPC_PLAGUED_RAT:
+                    case NPC_PLAGUED_INSECT:
+                    case NPC_PLAGUED_MAGGOT:
+                    {
+                        auto el = PlaguedGatesGUIDs.find(who->GetGUID());
+                        uint8 uiGate = el->second;
```

Comment:
You are possibly dereferencing invalid (.end()) iterator here, all of the code here should be under `if (el != PlaguedGatesGUIDs.end())`, not just erase